src/async_process.rs

```diff
@@ -0,0 +1,182 @@
+use std::task::{Context, Poll};
+
+cfg_if::cfg_if! {
+    if #[cfg(feature = "async-std-runtime")] {
+        use ::async_std::process;
+    } else if #[cfg(feature = "tokio-runtime")] {
+        use ::tokio::process;
+    }
+}
+
+#[derive(Debug)]
+pub struct Command {
+    inner: process::Command,
+}
+
+impl Command {
+    pub fn new<S: AsRef<OsStr>>(program: S) -> Self {
+        let inner = process::Command::new(program);
+        Self { inner }
+    }
+
+    pub fn arg<S: AsRef<OsStr>>(&mut self, arg: S) -> &mut Self {
+        self.inner.arg(arg);
+        self
+    }
+
+    pub fn args<I, S>(&mut self, args: I) -> &mut Self
+    where
+        I: IntoIterator<Item = S>,
+        S: AsRef<OsStr>,
+    {
+        self.inner.args(args);
+        self
+    }
+
+    pub fn envs<I, K, V>(&mut self, vars: I) -> &mut Self
+    where
+        I: IntoIterator<Item = (K, V)>,
+        K: AsRef<OsStr>,
+        V: AsRef<OsStr>,
+    {
+        self.inner.envs(vars);
+        self
+    }
+
+    pub fn stderr<T: Into<Stdio>>(&mut self, cfg: T) -> &mut Self {
+        self.inner.stderr(cfg);
+        self
+    }
+
+    pub fn spawn(&mut self) -> std::io::Result<Child> {
+        let inner = self.inner.spawn()?;
+        Ok(Child::new(inner))
+    }
+}
+
+#[derive(Debug)]
+pub struct Child {
+    pub stderr: Option<ChildStderr>,
+    pub inner: process::Child,
+}
+
+/// Wrapper for an async child process.
+///
+/// The inner implementation depends on the selected async runtime (features `async-std-runtime`
+/// or `tokio-runtime`).
+impl Child {
+    fn new(mut inner: process::Child) -> Self {
+        let stderr = inner.stderr.take();
+        Self {
+            inner,
+            stderr: stderr.map(|inner| ChildStderr { inner }),
+        }
+    }
+
+    /// Kill the child process, asynchronously if possible (otherwise by blocking)
+    ///
+    /// - `async-std-runtime`: blocking call to `async_std::process::Child::kill`.
+    /// - `tokio-runtime`: async call to `tokio::process::Child::kill`
+    pub async fn async_kill(&mut self) -> std::io::Result<()> {
+        cfg_if::cfg_if! {
+            if #[cfg(feature = "async-std-runtime")] {
+                self.inner.kill()
+            } else if #[cfg(feature = "tokio-runtime")] {
+                self.inner.kill().await
+            }
+        }
+    }
+
+    /// Kill the child process synchronously (blocking)
+    ///
+    /// - `async-std-runtime`: blocking call to `async_std::process::Child::kill`.
+    /// - `tokio-runtime`: async call to `tokio::process::Child::kill`, resolved with `tokio::runtime::Handle::block_on`
+    pub fn kill(&mut self) -> std::io::Result<()> {
+        cfg_if::cfg_if! {
+            if #[cfg(feature = "async-std-runtime")] {
+                self.inner.kill()
+            } else if #[cfg(feature = "tokio-runtime")] {
+                let fut = self.async_kill();
+                let handle = tokio::runtime::Handle::current();
+                handle.block_on(fut)
```

Comment:
This is bad... It crashes the thread if called from an async thread.
I consider this a breaking change and I would have preferred if it was named differently.
```
thread 'actix-rt|system:0|arbiter:4' panicked at 'Cannot start a runtime from within a runtime. This happens because a function (like `block_on`) attempted to block the current thread while the thread is being used to drive asynchronous tasks.', C:\Users\sytten\.cargo\git\checkouts\dependency-chromiumoxide-fc437d36c499446b\17fba33\src\async_process.rs:106:24
```